vite-plugin/index.d.ts

```diff
@@ -30,6 +30,12 @@ interface QuasarPluginOpts {
   // runMode?: "web-client" | "ssr-client" | "ssr-server";
 }
 
-export function quasar(
-  opts?: QuasarPluginOpts
-): Plugin;
+export function quasar(opts?: QuasarPluginOpts): Plugin;
+
+export interface TransformAssetUrls {
```

Comment:
I think it would be better if this was imported directly from vue (since a vue package defines it)

Reply:
Hey, thanks for the info! Didn't know that.
"Sadly" the definitions are in `compiler-sfc` and I think it would be overkill to add this as a peer dependency just for this "simple" type. So I copied it over and added a link. I think this is a good compromise.

Reply:
If I recall correctly, compiler-sfc is now available as a nested import via `vue/compiler-sfc` or similar, since keeping those in sync for libraries was a pain. Dunno if they also export types from there, but worth taking a look

Reply:
I tried to import the type without adding a dependency, but it did not work / got a TS (or lint?) error

Reply:
That's strange, vue now imports `@vue/compiler-sfc` dependency, so you should have it no matter what

![image](https://user-images.githubusercontent.com/10036108/139840327-cc25bde1-385f-4084-a364-1778e2691f88.png)

Have you added `vue` and `vite` as devDependencies?
If you run `yarn -D vite vue@next`, you should be able to import the type as

```ts
import type { SFCTemplateCompileOptions } from "vue/compiler-sfc";

export type TransformAssetUrls = Required<SFCTemplateCompileOptions["transformAssetUrls"]>;
```

Not sure we support the whole type or only `AssetURLOptions`, haven't checked it